test/test_wolk.py

```diff
@@ -440,6 +417,281 @@ def test_add_simple_device(self):
+        with self.assertRaises(ValueError):
+            wolk = Wolk(  # noqa
+                "host",
+                1883,
+                "module_name",
+                lambda device_key: DeviceStatus.CONNECTED,
+                configuration_provider=False,
+            )
+
+    def test_bad_configuration_provider_callable(self):
+        """Test passing something that isn't callable raises ValueError."""
+        with self.assertRaises(ValueError):
+            wolk = Wolk(  # noqa
+                "host",
+                1883,
+                "module_name",
+                lambda device_key: DeviceStatus.CONNECTED,
+                configuration_provider=lambda a, b: a,
+            )
+
+    def test_missing_configuration_provider(self):
+        """Test passing an config handler but no provider raises an exception."""
+        with self.assertRaises(RuntimeError):
+            wolk = Wolk(  # noqa
+                "host",
+                1883,
+                "module_name",
+                lambda a: a,
+                connectivity_service=MockConnectivityService(),
+                configuration_handler=mock_configuration_handler,
+            )
+
+    def test_bad_firmware_handler(self):
+        """Test passing a bad firmware handler raises an exception."""
+        with self.assertRaises(ValueError):
+            wolk = Wolk(  # noqa
+                "host",
+                1883,
+                "module_name",
+                lambda a: a,
+                firmware_handler=False,
+            )
+
+    def test_bad_data_protocol(self):
+        """Test passing a bad data protocol raises an exception."""
+        with self.assertRaises(ValueError):
+            wolk = Wolk(  # noqa
+                "host",
+                1883,
+                "module_name",
+                lambda a: a,
+                data_protocol=False,
+            )
+
+    def test_good_data_protocol(self):
+        """Test passing a good data protocol doesn't raise an exception."""
+        data_protocol = JsonDataProtocol()
+
+        wolk = Wolk(
+            "host",
+            1883,
+            "module_name",
+            lambda a: a,
+            data_protocol=data_protocol,
+        )
+
+        self.assertEqual(data_protocol, wolk.data_protocol)
+
+    def test_bad_firmware_update_protocol(self):
+        """Test passing a bad firmware update protocol raises an exception."""
+        with self.assertRaises(ValueError):
+            wolk = Wolk(  # noqa
+                "host",
+                1883,
+                "module_name",
+                lambda a: a,
+                firmware_update_protocol=False,
+            )
+
+    def test_good_firmware_update_protocol(self):
+        """Test passing a good firmware update protocol doesn't raise an exception."""
+        firmware_update_protocol = JsonFirmwareUpdateProtocol()
+
+        wolk = Wolk(
+            "host",
+            1883,
```

Comment:
why test connection with unsecure port?

Reply:
No connection is established, this is just to satisfy the constructor